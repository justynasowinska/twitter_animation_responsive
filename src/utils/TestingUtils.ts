import { ShallowWrapper, ReactWrapper } from 'enzyme';

/**
 * Gets Test ID for component for testing purposes.
 * According to React Native documentation, setting testID can make performance worse,
 * so we can set these ids only in development environment.
 * https://facebook.github.io/react-native/docs/view#testid
 * @param {string} testId  Test id that should be set.
 * @returns {string | undefined}    If is PROD envirenment these id will be undefined.
 */
export const setTestID = (testId: string): string | undefined => {
    if (__DEV__) {
        return testId;
    }

    return undefined;
};

/**
 * Get ReactWrapper or ShallowWrapper that contains found nodes.
 * @param {ShallowWrapper<any, any, any> | ReactWrapper<any, any, any>}  wrapper     Wrapper got by shallow or mount method.
 * @param {string}                                                       testId   Test Id which contains wrapper that we want to find.
 * @returns {ShallowWrapper<any, any, any> | ReactWrapper<any, any, any>} Returns ShallowWrapper (for shallow method) or ReactWrapper (for mount method).
 */
export const findByTestID = (
    wrapper: ShallowWrapper<any, any, any> | ReactWrapper<any, any, any>,
    testId: string
    ): ShallowWrapper<any, any, any> | ReactWrapper<any, any, any> => {
        return wrapper.findWhere((node: ShallowWrapper<any, any, any> | ReactWrapper<any, any, any>) => node.prop('testID') === testId);
};