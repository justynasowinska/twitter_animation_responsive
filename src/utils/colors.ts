export const colors = {
    headerBgColor: '#1DA1F2',
    usernameDarkTextColor: '#14171a',
    usernameLightTextColor: '#ffffff',
    avatarBorderColor: '#ffffff',
    appBackgroundColor: '#ffffff'
};