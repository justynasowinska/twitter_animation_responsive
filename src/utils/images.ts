export const images = {
    avatarImg: require('../../assets/images/me.jpg')
};