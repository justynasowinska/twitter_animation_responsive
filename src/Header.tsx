import React from 'react';
import { StyleSheet, Animated } from 'react-native';
import { colors } from './utils/colors';
import { setTestID } from './utils/TestingUtils';

interface PropsType {
  animatedHeaderStyles: any;
  animatedTextStyles: any;
  userName: string;
  userNameFontSize: number;
}

export const Header = (props: PropsType) => {
    const { animatedHeaderStyles, animatedTextStyles, userName, userNameFontSize } = props;

    return (
        <Animated.View style={[styles.header, animatedHeaderStyles]}>
            <Animated.Text
              testID={setTestID('headerUserName')}
              style={
                [styles.username, { fontSize:  userNameFontSize }, animatedTextStyles]
                }
            >
              {userName}
            </Animated.Text>
        </Animated.View>
    );
}

const styles = StyleSheet.create({
  header: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    backgroundColor: colors.headerBgColor,
    alignItems: 'center',
  },
  usernameContainer: {
    position: 'absolute'
  },
  username: {
    position: 'absolute',
    fontWeight: 'bold',
    color: colors.usernameLightTextColor
  }
});
