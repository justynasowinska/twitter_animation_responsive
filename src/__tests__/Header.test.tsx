import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Header } from '../Header';
import { findByTestID } from '../utils/TestingUtils';

describe('Header component', () => {
    let component: ShallowWrapper;

    beforeEach(() => {
        component = shallow(
            <Header
                animatedHeaderStyles={undefined}
                animatedTextStyles={undefined}
                userName="Someone"
                userNameFontSize={12}
            />
        );
    });

    it('render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('renders given username', () => {
        const username = 'Wendy_W';
        component.setProps({ userName: username });

        expect(findByTestID(component, 'headerUserName').render().text()).toBe(username);
    });
});