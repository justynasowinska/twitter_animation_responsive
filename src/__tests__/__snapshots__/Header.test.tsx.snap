// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render correctly 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#1DA1F2",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontWeight": "bold",
          "position": "absolute",
        },
        Object {
          "fontSize": 12,
        },
        undefined,
      ]
    }
    testID="headerUserName"
  >
    Someone
  </AnimatedComponent>
</AnimatedComponent>
`;
